src/main/java/seedu/address/model/person/UniqueAssignmentList.java

```diff
@@ -0,0 +1,139 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.AssignmentNotFoundException;
+import seedu.address.model.person.exceptions.DuplicateAssignmentException;
+
+/**
+ * A list of assignments that enforces uniqueness between its elements and does not allow nulls.
+ * A assignment is considered unique by comparing using {@code Assignment#isSameAssignment(Assignment)}. As such,
+ * adding and updating of assignments uses Assignment#isSameAssignment(Assignment) for equality so as to ensure
+ * that the assignment being added or updated is unique in terms of identity in the UniqueAssignmentList.
+ * However, the removal of a assignment uses Assignment#equals(Object) so
+ * as to ensure that the assignment with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Assignment#isSameAssignment(Assignment)
+ */
+public class UniqueAssignmentList implements Iterable<Assignment> {
+
+    private final ObservableList<Assignment> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Assignment> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent assignment as the given argument.
+     */
+    public boolean contains(Assignment toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameAssignment);
+    }
+
+    /**
+     * Adds a assignment to the list.
+     * The assignment must not already exist in the list.
+     */
+    public void add(Assignment toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateAssignmentException();
+        }
+        internalList.add(toAdd);
+    }
```

Comment:
Since assignment starts with a vowel, it should be "an assignment" instead of "a assignment" :) The same typo can be found throughout the project. You can consider using the "find in path" feature in IntelliJ to help you find them (Edit -> Find -> Find in Path)